controllers/util/status.go

```diff
@@ -280,3 +292,23 @@ func MapNoobaaNegativeConditions(conditions *[]conditionsv1.Condition, found *nb
 	}
 
 }
+
+// SetVersionMismatchCondition sets the ConditionReconcileComplete to True and other Conditions
+// to indicate that the reconciliation process has completed successfully.
+func SetVersionMismatchCondition(conditions *[]conditionsv1.Condition, reason string, message string) {
```

Comment:
instead of doing if else we could get the status as argument IMO

Reply:
I thought that, but looking at other function definitions I opted for this approach.
Will update the function to include the status as an argument

Reply:
Though, I had one doubt if we want to show a "VerionMismatch" status always or not?
As per the current code, the status is updated only if the value is already set and if we want to keep it that way then this type of conditioning is needed.

Reply:
I have tweaked the code a bit, let me know if its better now.